eng/pipelines/templates/Static_Source_Analysis.yml

```diff
@@ -0,0 +1,82 @@
+parameters:
+  - name: isOfficialBuild
+    type: boolean
+    default: false
+
+steps:
+- task: PowerShell@1
+  displayName: "Print Environment Variables"
+  inputs:
+    scriptType: "inlineScript"
+    inlineScript: |
+      Get-ChildItem Env: | Sort-Object Name | Format-Table -Wrap -AutoSize
+
+- task: CredScan@2
+  inputs:
+    toolMajorVersion: "V2"
+  continueOnError: ${{ parameters.isOfficialBuild }}
+  condition: succeededOrFailed()
+
+- task: PowerShell@2
+  displayName: "Analyze CredScan artifact"
+  inputs:
+    targetType: 'inline'
+    script: |
+      try {
+        $credentialScannerMatches = Join-Path $(Pipeline.Workspace)\_sdt\logs\CredentialScanner CredentialScanner-matches.xml
+        $credScannerXml = [xml](Get-Content $credentialScannerMatches)
+
+        if($credScannerXml.DEFECTS -eq $null)
+        {
+          Write-Host "Unable to verify CodeAnalysisLogs artifact content, please manually verify it."
+          exit 1
+        }
+
+        if ($credScannerXml.DEFECTS.DEFECT -ne $null) {
+          Write-Host "CredentialScanner found following credential in this PR, please remove them from PR and use git squash/drop to remove commits with credential from git history then do force push:"
+          $credScannerXml.DEFECTS.DEFECT
+          exit 1
+        }
+      }
+      catch
+      {
+        Write-Host "##vso[task.LogIssue type=error;]$Error[0]"
+        exit 1
+      }
+  continueOnError: ${{ parameters.isOfficialBuild }}
+  condition: succeededOrFailed()
+
+- task: MSBuild@1
+  displayName: "Ensure msbuild.exe can parse nuget.sln"
+  inputs:
+    solution: "nuget.sln"
+    msbuildArguments: "/t:EnsureNewtonsoftJsonVersion"
+    maximumCpuCount: true
+  continueOnError: ${{ parameters.isOfficialBuild }}
+  condition: succeededOrFailed()
+
+- task: MSBuild@1
+  displayName: "Ensure package versions are declared in packages.targets"
+  inputs:
+    solution: "build\\build.proj"
+    msbuildArguments: "/t:EnsurePackageReferenceVersionsInSolution"
+  continueOnError: ${{ parameters.isOfficialBuild }}
+  condition: succeededOrFailed()
+
+- task: PowerShell@1
+  displayName: "Restore dotnet tools"
+  inputs:
+    scriptType: "inlineScript"
+    inlineScript: |
+      dotnet tool restore
+  continueOnError: ${{ parameters.isOfficialBuild }}
+  condition: succeededOrFailed()
+
+- task: PowerShell@1
+  displayName: "Check source file format"
```

Comment:
Only the NuGet.Commands project doesn't have whitespace analyzers due to https://github.com/NuGet/Home/issues/2345